In [ ]:
import pandas as pd
import plotly.express as px

In [ ]:
dataset = pd.read_csv("../data/speed_dating_data_unprocessed.csv", encoding='ISO-8859-1')
display(dataset.head())

iid   id  gender  idg  condtn  wave  round  position  positin1  order  ...  \
0    1  1.0       0    1       1     1     10         7       NaN      4  ...   
1    1  1.0       0    1       1     1     10         7       NaN      3  ...   
2    1  1.0       0    1       1     1     10         7       NaN     10  ...   
3    1  1.0       0    1       1     1     10         7       NaN      5  ...   
4    1  1.0       0    1       1     1     10         7       NaN      7  ...   

   attr3_3  sinc3_3  intel3_3  fun3_3  amb3_3  attr5_3  sinc5_3  intel5_3  \
0      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
1      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
2      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
3      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   
4      5.0      7.0       7.0     7.0     7.0      NaN      NaN       NaN   

   fun5_3  amb5_3  
0     NaN     NaN  
1     NaN     NaN  
2     NaN     NaN  
3     NaN     NaN  
4     NaN     NaN  

[5 rows x 195 columns]

---

## Does the main purpose of taking part in the event influence the decision to see a partner again ?

In [44]:
columns_to_keep = ["dec", "goal"]
df_goal = dataset[columns_to_keep].copy()

print("Missing values ratio (%) : ")
print(100 * df_goal.isnull().sum() / df_goal.shape[0])

Missing values ratio (%) : 
dec     0.000000
goal    0.942946
dtype: float64


In [45]:
df_goal = df_goal[columns_to_keep].dropna()
df_goal["goal"] = df_goal["goal"].map({1: "Seemed like a fun night out", 2: "To meet new people", 3: "To get a date", 
                                                       4: "Looking for a serious relationship", 5: "To say I did it", 
                                                       6: "Other"})

display(df_goal.head())

dec                goal
0    1  To meet new people
1    1  To meet new people
2    1  To meet new people
3    1  To meet new people
4    1  To meet new people

In [46]:
df_dec = df_goal.groupby("goal")["dec"].mean().reset_index()
df_dec = df_dec.sort_values(by="dec", ascending=False)


display(df_dec.head())

goal       dec
0  Looking for a serious relationship  0.521595
3                       To get a date  0.450079
4                  To meet new people  0.428287
2         Seemed like a fun night out  0.411559
1                               Other  0.396181

In [49]:
fig_2 = px.bar(
    df_dec,
    x="goal",
    y="dec",
    text="dec",
    color_discrete_sequence=px.colors.qualitative.G10
)

fig_2.update_layout(
    title="Decision ratio to go on second date according to purpose of event participation",
    xaxis_title="Purpose",
    yaxis_title="Decision ratio"
    
)

fig_2.update_traces(
    texttemplate="%{text:.2%}"
)

fig_2.show()

As a reminder, participants are mainly there to have fun (41%) or meet people (34%), with fewer looking for a date (7%) or a serious relationship (4%).

On this barplot, we can see that 52% of participants looking for a serious relationship wish to see their partner again, and although they are the least represented in the initial sample, their behavior has a notable influence on the results. For those who came simply for fun, 41% wanted to see their partner again.

In other words, the intention to enter into a serious relationship has a positive influence on the decision to see someone again, but participants with no particular intentions remain open to a second date.
